pkg/noderesourcetopology/filter.go

```diff
@@ -181,3 +207,21 @@ func resourceFoundOnNode(resName v1.ResourceName, wantQuantity resource.Quantity
 	}
 	return false
 }
+
+func allocateFromNUMA(nodes NUMANodeList, numaID int, container v1.Container) {
+	var nRes v1.ResourceList
+	// find the correct NUMA ID's resources
+	for i := 0; i < len(nodes); i++ {
+		if nodes[i].NUMAID == numaID {
+			nRes = nodes[i].Resources
+			break
+		}
+	}
+
+	cReq := container.Resources.Requests
+	for resName, quan := range cReq {
+		nodeResQuan := nRes[resName]
+		nodeResQuan.Sub(quan)
+		nRes[resName] = nodeResQuan
+	}
```

Comment:
This part can be moved inside the above foo loop:

```go
// Find the correct NUMA ID's resources and subtract from `nodes`.
func subtractFromNUMA(nodes NUMANodeList, numaID int, container v1.Container) {
	for i := 0; i < len(nodes); i++ {
		if nodes[i].NUMAID != numaID {
			continue
		}

		nRes := nodes[i].Resources
		for resName, quan := range container.Resources.Requests {
			nodeResQuan := nRes[resName]
			nodeResQuan.Sub(quan)
			nRes[resName] = nodeResQuan
		}
	}
}
```

Reply:
I don't have a strong preference here, so okay.